frame/identity/src/lib.rs

```diff
@@ -545,20 +576,38 @@ decl_module! {
 		/// identity.
 		///
 		/// - `subs`: The identity's sub-accounts.
+		/// - `old_subs_count`: This is the number of previous sub accounts associated with the identity.
 		///
 		/// # <weight>
-		/// - `O(S)` where `S` subs-count (hard- and deposit-bounded).
-		/// - At most two balance operations.
-		/// - At most O(2 * S + 1) storage mutations; codec complexity `O(1 * S + S * 1)`);
+		/// - `O(P + S)`
+		///   - where `P` old-subs-count (hard- and deposit-bounded).
+		///   - where `S` subs-count (hard- and deposit-bounded).
+		/// - At most one balance operations.
+		/// - At most O(P + S + 1) storage mutations; codec complexity `O(S)`);
 		///   one storage-exists.
+		/// - Benchmark: 115.2 + P * 5.11 + S * 6.67 µs (min square analysis)
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(50_000_000)]
-		fn set_subs(origin, subs: Vec<(T::AccountId, Data)>) {
+		#[weight = FunctionOf(
+			|(subs, &old_subs_count): (&Vec<(T::AccountId, Data)>, &u32)| {
+				let db = T::DbWeight::get();
+				db.reads(1) // storage-exists
+				+ db.reads_writes(1, 1) // balance ops
```

Comment:
This should be `(0,0)` since the sender account is the one who would have balance operations right?

Reply:
Can you elaborate on this?
Am I right that the sender's balance fields will be in cache already due to them sending the transaction so we don't expect another (effective) balance read_write?